capstone/src/main/java/com/google/sps/servlets/clubs/LabelsServlet.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/* Servlet that stores and returns data relating to clubs. */
+@WebServlet("/labels")
+public class LabelsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Map<String, Integer> labelFrequency = new HashMap<String, Integer>();
+    List<String> labels = new ArrayList<String>();
+
+    Query query = new Query(Constants.CLUB_ENTITY_PROP);
+    PreparedQuery prepared = datastore.prepare(query);
+
+    for (Entity entity : prepared.asIterable()) {
+      for (String label : ServletUtil.getPropertyList(entity, Constants.LABELS_PROP)) {
```

Comment:
I thought about this for a bit, but I couldn't wrap my head around how to do this as a stream, since it's a little bit more complicated with the nested loop. I'll try to think about this more...

Reply:
I think it could be simple if you transform into a flatMap of all the Labels. So something like (not including complete syntax):

prepared.asIterable().stream()
// this combines all the individual property lists from each entity
.flatMap(getPropertyList(enitity, LABELS_PROP).stream())
.forEach(label -> {if(labelFrequency) .... }) // rest of code can be contained in this lambda function

Reply:
Ohh, I see. It would probably also be good practice to extract out the "if (labelFrequency)..." part into a separate function, right?